hopsworks-api/src/main/java/io/hops/hopsworks/api/user/AuthService.java

```diff
@@ -183,6 +198,101 @@ public Response logout(@Context HttpServletRequest req) throws UserException, In
     return Response.ok().build();
   }
 
+  @POST
+  @Path("/service")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response serviceLogin(@FormParam("email") String email, @FormParam("password") String password,
+      @Context HttpServletRequest request)
+    throws LoginException, UserException, GeneralSecurityException, SigningKeyNotFoundException,
+      DuplicateSigningKeyException, HopsSecurityException {
+    if (Strings.isNullOrEmpty(email)) {
+      throw new IllegalArgumentException("Email cannot be null or empty");
+    }
+    if (Strings.isNullOrEmpty(password)) {
+      throw new IllegalArgumentException("Password cannot be null or empty");
+    }
+    Users user = userFacade.findByEmail(email);
+    if (user == null) {
+      throw new LoginException("Could not find registered user with email " + email);
+    }
+    if (!needLogin(request, user)) {
+      return Response.ok().build();
+    }
+    if (!userController.isUserInRole(user, "AGENT")) {
+      throw new HopsSecurityException(RESTCodes.SecurityErrorCode.REST_ACCESS_CONTROL, Level.FINE,
```

Comment:
Should we block `AGENT` from logging using the `/login` endpoint?

Reply:
I wouldn't. `AGENT` is still a user. In the future what we should do, is have a separate service account.